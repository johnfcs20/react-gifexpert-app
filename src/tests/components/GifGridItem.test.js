import React from "react"
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en <GifGridItem />', () => { 

    const tittle = 'un titulo';
    const url = 'https://localhost/algo.jpg'    
    const wrapper = shallow(<GifGridItem title={tittle} url={url} />)


    test('Debe de mostrar el texto correctamente', () => {         
        expect(wrapper).toMatchSnapshot();
     })

     test('Debe de mtener un parrafo con el tittle', () => { 
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(tittle);
     })

     test('Debe de tener una imagen igual a la de la URL y ALT igual a los Props.', () => { 
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(tittle);
     }) 

     test('Debe de tener animate__animated', () => { 
        const div = wrapper.find('div');
        expect(div.prop('className').includes('animate__animated')).toBe(true);
     })

 })