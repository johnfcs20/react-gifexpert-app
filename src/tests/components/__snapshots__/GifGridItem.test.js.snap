// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el texto correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
